components/purchase_modal/purchase_modal.tsx

```diff
@@ -560,9 +811,24 @@ class PurchaseModal extends React.PureComponent<Props, State> {
 
         const showPlanLabel =
                     this.state.selectedProduct?.sku ===
-                    CloudProducts.PROFESSIONAL;
+                    CloudProducts.PROFESSIONAL && !this.props.annualSubscription;
         const {formatMessage} = this.props.intl;
 
+        const findProductByID = (id: string) => {
+            return Object.values(this.props.products || {}).find(((product) => {
+                return product.id === id;
+            }));
+        };
+
+        const getYearlyPrice = () => {
+            if (!this.state.selectedProduct) {
+                return 0;
+            }
+
+            const crossSellsToProduct = findProductByID(this.state.selectedProduct.cross_sells_to);
+            return crossSellsToProduct ? crossSellsToProduct.price_per_seat / 12 : 0;
```

Comment:
The CWS backend already reports the monthly price to the frontend, so we should not need to do an adjustment here. https://github.com/mattermost/customer-web-server/blob/master/payment/stripe.go#L1964-L1967 . If local development is being done with mocks, it would be good to update those mocks locally. Also raised A PR that adds the mock to enterprise repo officially: https://github.com/mattermost/enterprise/pull/1296

Reply:
I don't think I'm using the mocked version, but I see the yearly product as this in redux: 
<img width="481" alt="image" src="https://user-images.githubusercontent.com/44761757/201755725-c1c555e5-7d71-4ee6-a119-1084c2583085.png">

And I don't have $90 as the price in my local `cloud_mock.go` file.

Reply:
Those are real ids for the product and price, and the price id its pulling was newly created to support changing the price from 120 to 90 for testing, so somehow its pulling all products even though test CWS doesn't return that product right now (I only get the monthly ones with `curl 'https://portal.test.cloud.mattermost.com/api/v1/products?family=cloud' | jq .`). The frontend talks to mattermost-server, and mattermost-server/enterprise talks to CWS. What are your server config values (e.g. local file, `mattermost-server/config/config.json`) for `CloudSettings.CWSAPIURL` and `CloudSettings.CWSURL`?

Reply:
Sorry, my bad, forgot to mention that I was using the temporary solution that Nick provided to get the cloud yearly products. But does the yearly product shown above look the way it's suppose to look (as in, is the price_per_seat for yearly products shown on a per year basis)

Reply:
@johndavidlugtu @annelieseklein Nathaniel and Emmy bring up a good point here - how do you want the yearly prices to be displayed to the customer?

The designs don't specify /user/month (but then also say $10, which would be the /user/month price of yearly) so it's unclear. 

The CWS returns the price as the full price for the year (ie, not /user/month, just /user/year)

Reply:
I misread the enterprise logic. Its currently returning the price for the yearly product as a year price, so the divide by 12 is correct.